src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -156,16 +160,18 @@ public static Configuration getBigQueryConfig(@Nullable String serviceAccountFil
    * @param storage the storage client for the project
    * @param datasetName the name of the dataset
    * @param bucketName the name of the bucket
+   * @param cmekKey the name of the cmek key
    * @throws IOException if there was an error creating or fetching any GCP resource
    */
   public static void createResources(BigQuery bigQuery, Storage storage,
-                                     String datasetName, String bucketName) throws IOException {
+                                     String datasetName, String bucketName,
+                                     @Nullable String cmekKey) throws IOException {
     Dataset dataset = bigQuery.getDataset(datasetName);
     Bucket bucket = storage.get(bucketName);
 
     if (dataset == null && bucket == null) {
       createBucket(storage, bucketName, null,
-                   () -> String.format("Unable to create Cloud Storage bucket '%s'", bucketName));
+                   () -> String.format("Unable to create Cloud Storage bucket '%s'", bucketName), cmekKey);
```

Comment:
it's easier to read these method calls if the cmekKey comes before the Supplier, so that the closure is the last argument and not in the middle